contributors/devel/e2e-tests.md

```diff
@@ -136,6 +136,14 @@ with this command:
 go run hack/e2e.go -- -v --down
 ```
 
+## Building the Tests
+
+* You can quickly recompile the e2e testing framework via `go install ./test/e2e`.
+  This will not do anything besides allow you to verify that the go code compiles.
+
+* If you want to run your e2e testing framework without re-provisioning the e2e setup,
```

Comment:
Do you mean `make WHAT=test/e2e/e2e.test`? It rebuilds all e2e test tools, where as `go install ./test/e2e` just compiles the e2e framework, and `go run hack/e2e.go -- --test` runs the tests (but does not recompile the tools or the framework).

Example outputs:

```
xxx: ~/.../src/k8s.io/kubernetes [1] (ingress-pre-shared-tls u= origin/ingress-pre-shared-tls)
make WHAT=test/e2e/e2e.test
+++ [0323 19:19:41] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [0323 19:19:42] Generating bindata:
    test/e2e/generated/gobindata_util.go
~/Code/go/src/k8s.io/kubernetes ~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
+++ [0323 19:19:43] Building go targets for darwin/amd64:
    cmd/libs/go2idl/deepcopy-gen
+++ [0323 19:19:51] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [0323 19:19:51] Generating bindata:
    test/e2e/generated/gobindata_util.go
~/Code/go/src/k8s.io/kubernetes ~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
+++ [0323 19:19:52] Building go targets for darwin/amd64:
    cmd/libs/go2idl/defaulter-gen
+++ [0323 19:19:59] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [0323 19:19:59] Generating bindata:
    test/e2e/generated/gobindata_util.go
~/Code/go/src/k8s.io/kubernetes ~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
+++ [0323 19:20:00] Building go targets for darwin/amd64:
    cmd/libs/go2idl/conversion-gen
+++ [0323 19:20:08] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [0323 19:20:08] Generating bindata:
    test/e2e/generated/gobindata_util.go
~/Code/go/src/k8s.io/kubernetes ~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
+++ [0323 19:20:09] Building go targets for darwin/amd64:
    cmd/libs/go2idl/openapi-gen
+++ [0323 19:20:17] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
    k8s.io/kubernetes/vendor/github.com/jteeuwen/go-bindata/go-bindata
+++ [0323 19:20:17] Generating bindata:
    test/e2e/generated/gobindata_util.go
~/Code/go/src/k8s.io/kubernetes ~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
~/Code/go/src/k8s.io/kubernetes/test/e2e/generated
+++ [0323 19:20:18] Building go targets for darwin/amd64:
    test/e2e/e2e.test
# k8s.io/kubernetes/pkg/apis/extensions/validation
pkg/apis/extensions/validation/validation.go:392: undefined: extensions.IngressPreSharedCertAnnotationKey
# k8s.io/kubernetes/pkg/printers/internalversion
pkg/printers/internalversion/printers.go:798: undefined: extensions.IngressPreSharedCertAnnotationKey
!!! [0323 19:21:30] Call tree:
!!! [0323 19:21:30]  1: /xxx/Code/go/src/k8s.io/kubernetes/hack/lib/golang.sh:740 kube::golang::build_binaries_for_platform(...)
!!! [0323 19:21:30]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make: *** [all] Error 1
```
```
xxx: ~/.../src/k8s.io/kubernetes [2] (ingress-pre-shared-tls u= origin/ingress-pre-shared-tls)
go install ./test/e2e
# k8s.io/kubernetes/pkg/apis/extensions/validation
pkg/apis/extensions/validation/validation.go:392: undefined: extensions.IngressPreSharedCertAnnotationKey
# k8s.io/kubernetes/pkg/printers/internalversion
pkg/printers/internalversion/printers.go:798: undefined: extensions.IngressPreSharedCertAnnotationKey
```
```
xxx: ~/.../src/k8s.io/kubernetes [1] (ingress-pre-shared-tls u= origin/ingress-pre-shared-tls)
go run hack/e2e.go -- -v --test
2017/03/23 19:24:52 e2e.go:76: Calling kubetest -v --test...
2017/03/23 19:24:52 util.go:108: Running: ./cluster/kubectl.sh version --match-server-version=false
Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.1.992+1ad5cea24e7d85", GitCommit:"1ad5cea24e7d85974c6f6519187d43ecd384e5ae", GitTreeState:"clean", BuildDate:"2017-02-15T20:45:30Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
2017/03/23 19:24:52 util.go:110: Step './cluster/kubectl.sh version --match-server-version=false' finished in 350.573923ms
2017/03/23 19:24:52 util.go:108: Running: ./hack/e2e-internal/e2e-status.sh
Project: xxx
Zone: us-central1-b
Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.1.992+1ad5cea24e7d85", GitCommit:"1ad5cea24e7d85974c6f6519187d43ecd384e5ae", GitTreeState:"clean", BuildDate:"2017-02-15T20:45:30Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
2017/03/23 19:24:54 util.go:110: Step './hack/e2e-internal/e2e-status.sh' finished in 1.338509916s
2017/03/23 19:24:54 util.go:108: Running: ./hack/ginkgo-e2e.sh
Setting up for KUBERNETES_PROVIDER="gce".
```